samples/s02-instances/hostCode.cpp

```diff
@@ -0,0 +1,342 @@
+  GPRTContext context = gprtContextCreate();
+  GPRTModule module = gprtModuleCreate(context,s02_deviceCode);
+
+  // ##################################################################
+  // set up all the GPU kernels we want to run
+  // ##################################################################
+
+  // -------------------------------------------------------
+  // declare geometry type
+  // -------------------------------------------------------
+  GPRTVarDecl trianglesGeomVars[] = {
+    { "index",  GPRT_BUFFER, GPRT_OFFSETOF(TrianglesGeomData,index)},
+    { "vertex", GPRT_BUFFER, GPRT_OFFSETOF(TrianglesGeomData,vertex)},
+    { /* sentinel to mark end of list */ }
+  };
+  GPRTGeomType trianglesGeomType
+    = gprtGeomTypeCreate(context,
+                        GPRT_TRIANGLES,
+                        sizeof(TrianglesGeomData),
+                        trianglesGeomVars,-1);
+  gprtGeomTypeSetClosestHitProg(trianglesGeomType,0,
+                           module,"TriangleMesh");
+
+  // -------------------------------------------------------
+  // set up miss prog
+  // -------------------------------------------------------
+  GPRTVarDecl missProgVars[]
+    = {
+    { "color0", GPRT_FLOAT3, GPRT_OFFSETOF(MissProgData,color0)},
+    { "color1", GPRT_FLOAT3, GPRT_OFFSETOF(MissProgData,color1)},
+    { /* sentinel to mark end of list */ }
+  };
+  GPRTMiss miss
+    = gprtMissCreate(context,module,"miss",sizeof(MissProgData),
+                        missProgVars,-1);
+
+  // -------------------------------------------------------
+  // set up ray gen program
+  // -------------------------------------------------------
+  GPRTVarDecl rayGenVars[] = {
+    { "fbSize",        GPRT_INT2,   GPRT_OFFSETOF(RayGenData,fbSize)},
+    { "fbPtr",         GPRT_BUFFER, GPRT_OFFSETOF(RayGenData,fbPtr)},
+    { "world",         GPRT_ACCEL,  GPRT_OFFSETOF(RayGenData,world)},
+    { "camera.pos",    GPRT_FLOAT3, GPRT_OFFSETOF(RayGenData,camera.pos)},
+    { "camera.dir_00", GPRT_FLOAT3, GPRT_OFFSETOF(RayGenData,camera.dir_00)},
+    { "camera.dir_du", GPRT_FLOAT3, GPRT_OFFSETOF(RayGenData,camera.dir_du)},
+    { "camera.dir_dv", GPRT_FLOAT3, GPRT_OFFSETOF(RayGenData,camera.dir_dv)},
+    { /* sentinel to mark end of list */ }
+  };
+
+  GPRTRayGen rayGen
+    = gprtRayGenCreate(context,module,"simpleRayGen",
+                      sizeof(RayGenData),
+                      rayGenVars,-1);
+
+  // ##################################################################
+  // set the parameters for those kernels
+  // ##################################################################
+
+  // Setup pixel frame buffer
+  GPRTBuffer frameBuffer
+    = gprtDeviceBufferCreate(context,GPRT_INT,fbSize.x*fbSize.y);
+  gprtRayGenSetBuffer(rayGen,"fbPtr", frameBuffer);
+  gprtRayGenSet2iv(rayGen,"fbSize", (int32_t*)&fbSize);
+
+  // Miss program checkerboard background colors
+  gprtMissSet3f(miss,"color0",0.1f,0.1f,0.1f);
```

Comment:
Again, a really small style thing, but there are places where args have a space after the comma and other places where they don't. Picking a style and being consistent about it now will help us down the road.

Reply:
finding that I can autoformat with clang format in a vscode extension. Should hopefully not be an issue now.